test/e2etest.py

```diff
@@ -1325,6 +1335,66 @@ def test_delete_plan(sample_user, sample_team, sample_template_name, sample_appl
     assert 'incidents have been created using it' in re.json()['title']
 
 
+def test_restricted_plan(sample_non_admin_user, sample_admin_user, sample_team, sample_template_name):
+
+    data = {
+        'creator': sample_admin_user,
+        'name': 'restricted',
+        'description': 'Test plan for e2e test',
+        'threshold_window': 900,
+        'threshold_count': 10,
+        'aggregation_window': 300,
+        'aggregation_reset': 300,
+        'steps': [
+            [
+                {
+                    'role': 'team',
+                    'target': sample_team,
+                    'priority': 'low',
+                    'wait': 600,
+                    'repeat': 0,
+                    'template': sample_template_name
+                },
+            ],
+        ],
+        'isValid': True
+    }
+
+    # Test creating and deleting by ID
+    re = requests.post(base_url + 'plans', json=data, headers=username_header(sample_admin_user))
+    assert re.status_code == 201
+    plan_id = int(re.content.strip())
+    assert plan_id
+
+    # test creating a restricted plan
+    data = {
+        'creator': sample_non_admin_user,
+        'name': 'restricted',
+        'description': 'Test plan for e2e test',
+        'threshold_window': 900,
+        'threshold_count': 10,
+        'aggregation_window': 300,
+        'aggregation_reset': 300,
+        'steps': [
+            [
+                {
+                    'role': 'team',
+                    'target': sample_team,
+                    'priority': 'low',
+                    'wait': 600,
+                    'repeat': 0,
+                    'template': sample_template_name
+                },
+            ],
+        ],
+        'isValid': True
+    }
+
+    # Test creating and deleting by ID
+    re = requests.post(base_url + 'plans', json=data, headers=username_header(sample_non_admin_user))
+    assert re.status_code == 401
```

Comment:
Can we confirm the received error message is what we expect, to make sure we're getting a 401 for the intended reason?

Reply:
added